lib/vtk/commands/socks/setup.rb

```diff
@@ -0,0 +1,326 @@
+          else
+            output.puts ' ❌ ERROR: SSH Connection to SOCKS server unsuccessful. Error message:'
+            output.puts `ssh -F #{ssh_config_path} -o ConnectTimeout=5 -vvv socks -D #{port} -N`
+            exit 1
+          end
+        end
+
+        def add_ip_to_known_hosts
+          jump_box_ip = `grep -A 2 'Host socks' ~/.ssh/config | grep ProxyCommand | awk '{print $6}'`.chomp
+          socks_ip = `grep -A 2 'Host socks' ~/.ssh/config | grep HostName | awk '{print $2}'`.chomp
+
+          `ssh-keygen -R #{jump_box_ip}` if File.exist? '~/.ssh/known_hosts'
+          `ssh-keygen -R #{socks_ip}` if File.exist? '~/.ssh/known_hosts'
+          `ssh-keyscan -H #{jump_box_ip} >> ~/.ssh/known_hosts 2> /dev/null`
+          `ssh -i #{ssh_key_path} dsva@#{jump_box_ip} 'ssh-keyscan -H #{socks_ip}' >> ~/.ssh/known_hosts 2> /dev/null`
+        end
+
+        def configure_system_boot
+          return false unless macos?
+          return true unless `launchctl list | grep #{launch_agent_label}`.empty?
+          return false unless prompt.yes? '----> Start SOCKS on system boot?'
+
+          install_autossh
+          install_launch_agent
+        end
+
+        def launch_agent_label
+          @launch_agent_label ||= begin
+            launch_agent_label = 'gov.va.socks'
+            launch_agent_label += "-test-#{rand 1000}" if ENV['TEST'] == 'test'
+            launch_agent_label
+          end
+        end
+
+        def install_autossh
+          installed = !`command -v autossh`.empty?
+          return true if installed
+
+          return false unless prompt.yes? '----> Autossh missing. Install now via brew?'
+
+          `brew install autossh`
+        end
+
+        def install_launch_agent
+          unless File.exist? "#{boot_script_path}/LaunchAgents/gov.va.socks.plist"
+            FileUtils.mkdir_p "#{boot_script_path}/Logs/autossh"
+            FileUtils.mkdir_p "#{boot_script_path}/LaunchAgents"
+
+            write_launch_agent
+          end
+
+          `launchctl load -w #{boot_script_path}/LaunchAgents/gov.va.socks.plist`
+        end
+
+        def write_launch_agent
+          erb_template = File.read "#{__dir__}/gov.va.socks.plist.erb"
+          erb = ERB.new erb_template
+          launch_agent_contents = erb.result(
+            launch_agent_variables.instance_eval { binding }
+          )
+          File.write "#{boot_script_path}/LaunchAgents/gov.va.socks.plist", launch_agent_contents
+        end
+
+        def launch_agent_variables
+          OpenStruct.new(
+            label: launch_agent_label,
+            autossh_path: `which autossh`.chomp,
+            port: @port,
+            boot_script_path: File.realpath(boot_script_path),
+            user: ENV['USER']
+          )
+        end
+
+        def configure_system_proxy
+          return false unless macos?
+          return log 'Skipping system proxy configuration as custom --port was used.' unless port == '2001'
```

Comment:
The proxy.pac file has the port hard coded so `configure_system_proxy` doesn't work if user specifies a custom port.

Reply:
Why offer the custom port flag then?

Reply:
Partly for testing, partly in case they just want to set up socks key/config without doing the system proxy stuff. I could switch it to an ENV var. /shrug